packs/equipment-effects/effect-eagle-eye-elixir.json

```diff
@@ -0,0 +1,174 @@
+                        ],
+                        "value": "Compendium.pf2e.equipment-srd.Item.Eagle Eye Elixir (Lesser)"
+                    },
+                    {
+                        "predicate": [
+                            {
+                                "or": [
+                                    "parent:origin:item:slug:eagle-eye-elixir-moderate",
+                                    {
+                                        "not": "parent:origin:item"
+                                    }
+                                ]
+                            }
+                        ],
+                        "value": "Compendium.pf2e.equipment-srd.Item.Eagle Eye Elixir (Moderate)"
+                    },
+                    {
+                        "predicate": [
+                            {
+                                "or": [
+                                    "parent:origin:item:slug:eagle-eye-elixir-greater",
+                                    {
+                                        "not": "parent:origin:item"
+                                    }
+                                ]
+                            }
+                        ],
+                        "value": "Compendium.pf2e.equipment-srd.Item.Eagle Eye Elixir (Greater)"
+                    },
+                    {
+                        "predicate": [
+                            {
+                                "or": [
+                                    "parent:origin:item:slug:eagle-eye-elixir-major",
+                                    {
+                                        "not": "parent:origin:item"
+                                    }
+                                ]
+                            }
+                        ],
+                        "value": "Compendium.pf2e.equipment-srd.Item.Eagle Eye Elixir (Major)"
+                    }
+                ],
+                "key": "ChoiceSet",
+                "prompt": "PF2E.SpecificRule.Prompt.EquipmentVariety",
+                "rollOption": "eagle-eye-elixir"
+            },
+            {
+                "domain": "all",
+                "key": "RollOption",
+                "label": "PF2E.SpecificRule.Equipment.EagleEyeElixir.RollOptionLabel",
+                "option": "eagle-eye-find-hidden",
+                "toggleable": true
+            },
+            {
+                "key": "FlatModifier",
+                "selector": "perception",
+                "slug": "eagle-eye-elixir-bonus",
+                "type": "item",
+                "value": 1
+            },
+            {
+                "key": "AdjustModifier",
+                "mode": "upgrade",
+                "predicate": [
+                    "eagle-eye-find-hidden",
+                    "eagle-eye-elixir:eagle-eye-elixir-lesser"
+                ],
+                "selector": "perception",
+                "slug": "eagle-eye-elixir-bonus",
+                "value": 2
+            },
```

Comment:
You can do a late priority add instead to increase the modifier for hidden doors and traps. Plus make the predicate an `or` for both hidden doors and `origin:trait:trap`